### PR TITLE
CB-18759 Add Salt Update flow as the first in the upgrade CCM flowchain

### PR DESCRIPTION
**Pull request by bergerdenes**
As it turned out, only the Salt states were pushed + _only_ the gateway pillar. If recent Salt states would require new pillar data then those states would fail. Typically, at calling `highstate`.

Adjusted integration tests. Removed refactored service which is not in use anymore.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/chain/UpgradeCcmFlowEventChainFactory.java [hunk 1]
```diff
@@ -26,6 +28,7 @@ public String initEvent() {
     @Override
     public FlowTriggerEventQueue createFlowTriggerEventQueue(UpgradeCcmFlowChainTriggerEvent event) {
         Queue<Selectable> flowEventChain = new ConcurrentLinkedQueue<>();
+        flowEventChain.add(new StackEvent(SaltUpdateEvent.SALT_UPDATE_EVENT.event(), event.getResourceId(), event.accepted()));
         flowEventChain.add(
                 new UpgradeCcmTriggerRequest(UPGRADE_CCM_EVENT.event(), event.getResourceId(), event.getClusterId(), event.getOldTunnel(), event.accepted()));
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/chain/UpgradeCcmFlowEventChainFactory.java:33]: Is `event.accepted()` still needed here?

**Review by lajosrodek — CHANGES_REQUESTED** [review]

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/ccm/upgrade/UpgradeCcmService.java [hunk 2]
```diff
@@ -145,6 +150,15 @@ public void updateTunnel(Long stackId) {
         stackService.setTunnelByStackId(stackId, Tunnel.latestUpgradeTarget());
     }
 
+    public void pushSaltState(Long stackId, Long clusterId) {
+        InMemoryStateStore.putStack(stackId, PollGroup.POLLABLE);
+        if (clusterId != null) {
+            InMemoryStateStore.putCluster(clusterId, PollGroup.POLLABLE);
+        }
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/ccm/upgrade/UpgradeCcmService.java:157]: why is this necessary? usually stack state changes does the same

Comment (lacikaaa) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/ccm/upgrade/UpgradeCcmService.java:157]: I haven't asked to remove and retest. Although I see now it was moved from another class, which was deleted. I just don't understand why is it necessary at all.

**Pull request merged**